src/Exceptions/AblyException.php

```diff
@@ -1,22 +1,38 @@
 <?php
 namespace Ably\Exceptions;
 
-use Exception;
+use Ably\Models\ErrorInfo;
+use \Exception;
 
 /**
- * Generic exception for Ably classes
+ * Generic Ably exception
  */
 class AblyException extends Exception {
 
-	protected $ablyCode;
-    
-    public function __construct($message, $code = 400, $ablyCode = 40000) {
-        parent::__construct($message, $code);
+    /**
+     * @var ErrorInfo
+     */
+    public $errorInfo;
 
-        $this->ablyCode = $ablyCode;
+    public function __construct( $message, $code = 40000, $statusCode = 400 ) {
```

Comment:
I'm not sure if we need to specify a default `code` and `statusCode`.

Java needs all three (message + code + statusCode) to be present https://github.com/ably/ably-java/blob/master/src/io/ably/types/AblyException.java#L37 and Ruby uses `nil` by default.

If you can use `null` here as the default it would be nice.